controllers/poddistruptionbudget.go

```diff
@@ -63,24 +63,18 @@ func (r *SingleClusterReconciler) deletePDB() error {
 }
 
 func (r *SingleClusterReconciler) createOrUpdatePDB() error {
-	podList, err := r.getClusterPodList()
-	if err != nil {
-		return err
-	}
-
-	for podIdx := range podList.Items {
-		pod := &podList.Items[podIdx]
-
-		for containerIdx := range pod.Spec.Containers {
-			if pod.Spec.Containers[containerIdx].Name != asdbv1.AerospikeServerContainerName {
-				continue
-			}
+	// Check for cluster readiness status only when it's false.
```

Comment:
Shouldn't we rely on flag update by updateStatus flow and just make a following check here?
```
if !r.aeroCluster.Status.IsClusterReadinessEnabled {
	r.Log.Info("Pod Readiness is not enabled throughout cluster. Skipping PodDisruptionBudget." +
		" Refer Aerospike documentation for more details.")
	return nil
}
We can rely on

Reply:
In that case, PDB won't be updated in the same flow but it will get created in the next reconcile. I think reconcile runs one more time after updating the status.

Reply:
With that flag check, PBD creation was skipped as that flag is updated in the updateStatus and there is no reconcile after that. So changed that logic and optimised with false check